float64 dis_x
float64 dis_y
float64 theta
int16    mode
